File: server/src/main/java/de/ketrwu/portfolio/PortfolioApplication.java
package de.ketrwu.portfolio;

import de.ketrwu.portfolio.service.CaptchaService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableScheduling;

@SpringBootApplication
@EnableAutoConfiguration
@Configuration
@EnableScheduling
@ComponentScan("de.ketrwu.portfolio")
public class PortfolioApplication {

	@Autowired
	private CaptchaService captchaService;

	public static void main(String[] args) {
		PortfolioApplication portfolioApplication = new PortfolioApplication();
		SpringApplication.run(PortfolioApplication.class, args)
				.getAutowireCapableBeanFactory()
				.autowireBean(portfolioApplication);
		portfolioApplication.init();
	}

	public void init() {
		captchaService.loadFonts();
	}

}

File: server/src/main/kotlin/de/ketrwu/portfolio/PortfolioApplication.kt
package de.ketrwu.portfolio

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer
import org.springframework.scheduling.annotation.EnableScheduling

/**
 * SpringBootServletInitializer to init application when running as deployed WAR
 * @author Kenneth Wußmann
 */
@SpringBootApplication(scanBasePackages = ["de.ketrwu.portfolio"])
@EnableScheduling
open class PortfolioApplication : SpringBootServletInitializer() {

    override fun configure(builder: SpringApplicationBuilder?): SpringApplicationBuilder? {
        return builder?.sources(PortfolioApplication::class.java)
    }
}

/**
 * Main entry-point for runnable JAR
 */
@Suppress("SpreadOperator")
fun main(args: Array<String>) {
    SpringApplication.run(PortfolioApplication::class.java, *args)
}